src/components/Trade/Components/TradeQuotes/TradeQuote.tsx

```diff
@@ -161,6 +159,27 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
     return borderColor
   })()
 
+  const protocolIcon = useMemo(() => {
+    switch (swapperWithMetadata.swapper.getType()) {
+      case SwapperType.Osmosis:
+        return OsmosisIcon
+      case SwapperType.LIFI:
+        return LiFiIcon
+      case SwapperType.CowSwap:
+        return CowIcon
+      case SwapperType.ZrxAvalanche:
+      case SwapperType.ZrxBnbSmartChain:
+      case SwapperType.ZrxEthereum:
+      case SwapperType.ZrxOptimism:
+        return ZrxIcon
+      case SwapperType.Thorchain:
+        return THORChainIcon
+      case SwapperType.Test:
+      default:
+        return ''
+    }
+  }, [swapperWithMetadata])
```

Comment:
If we want to ensure that all possible `SwapperType`s are always covered:

```suggestion
  const protocolIcon = useMemo(() => {
    const swapperType = swapperWithMetadata.swapper.getType()
    switch (swapperType) {
      case SwapperType.Osmosis:
        return OsmosisIcon
      case SwapperType.LIFI:
        return LiFiIcon
      case SwapperType.CowSwap:
        return CowIcon
      case SwapperType.ZrxAvalanche:
      case SwapperType.ZrxBnbSmartChain:
      case SwapperType.ZrxEthereum:
      case SwapperType.ZrxOptimism:
        return ZrxIcon
      case SwapperType.Thorchain:
        return THORChainIcon
      case SwapperType.Test:
        return ''
      default:
        assertUnreachable(swapperType)
    }
  }, [swapperWithMetadata])
```